File: client/src/Types/Invoice.ts
export interface InvoiceItem {
  ProductName: string;
  Price: number;
  Quantity: number;
}
export interface Invoice {
  IsnewCustomer: boolean
  customerId?: number;
  invoiceDate?: string; 
  notes?: string;
  createdBy?: string;
  items: InvoiceItem[];
}
export interface FullInvoiceRequest {
  isNewCustomer: boolean;
  customer?: CustomerDto; 
  invoice: InvoiceDto;
}
export interface InvoiceDto {
  customerId: number;
  invoiceDate?: Date; 
  notes?: string;
  createdBy?: string;
  items: InvoiceItemDto[];
}
export interface InvoiceItemDto {
  productName: string;
  price: number;
  quantity: number;
} 
export interface CustomerDto {
  customerId: number;  
  name: string;          
  email?: string;
  phone?: string;
  billingAddress?: string;
  createdAt?: Date;
  invoices?: Invoices[]; 
}
export interface Invoices {
  invoiceId: number;
}

File: src/ERP.Modules.Retail/Domain/Invoice.cs
using ERP.Modules.Shared.Contracts;
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using ERP.Modules.Retail.Domain;
using ERP.Modules.Retail.Application;
using ERP.Modules.Retail.Contracts;
using ERP.Modules.Retail.Controllers;
using ERP.Modules.Retail.DTOs;
using ERP.Modules.Retail.Infrastructure;
using ERP.Modules.Retail.Enums;

namespace ERP.Modules.Retail.Domain
{
    public class Invoice : ITenantEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int InvoiceId { get; set; }
        public string InvoiceNumber { get; set; } = string.Empty;

        [ForeignKey(nameof(Customer))]
        public int CustomerId { get; set; }
        public DateTime InvoiceDate { get; set; }
        public decimal Subtotal { get; set; }
        public decimal TaxAmount { get; set; }
        public decimal TotalAmount { get; set; }
        public string? Notes { get; set; }
        public string? CreatedBy { get; set; }
        public DateTime CreatedAt { get; set; }
        public Customer? Customer { get; set; }
        public ICollection<InvoiceItem>? Items { get; set; }

        public int CompanyId { get; set; }
    }
}
